### PR TITLE
Borrel reservations

### LINKED ISSUE #169: Add reservations application
We should add a reservations application such that people with a BBB (BasisBorrelBrevet) are able to reserve a canteen and drinks/food.

### PR DESCRIPTION
**Pull request by KiOui**
This pull request adds a `borrel` application to TOSTI which replaces the functions of the [VoorraadCie](https://voorraadcie.nl/) website. The functions of the application include:

- Keeping track of people with a BasicBorrelBrevet
- Enabling people with a BasicBorrelBrevet to create ReservationRequests
- Enabling VoorraadCie members to respond to these ReservationRequests
- Enabling people with a BasicBorrelBrevet to track their ReservationRequests and see whether they are accepted
- Enabling people with a BasicBorrelBrevet to digitally hand in a BorrelForm
- Enabling VoorraadCie members to add and remove Products from the BorrelForm and their inventory

Closes #169

### CODE REVIEW

**Comment by KiOui** [comment]
This needs to be changed as a whole when #174 is merged.

File: website/borrel/selectors.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+def available_products():
+    """Get products that are available for order."""
+    return models.Product.objects.filter(active=True)
+
+
+def product_categories():
+    """Get all categories."""
+    return models.ProductCategory.objects.all()
```
Comment (JobDoesburg) [inline website/borrel/selectors.py:11]: What is this? It seems like you're trying to create your own ProductManager and CategoryManager (which is nice, but just build it as such then)

File: website/venues/selectors.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from venues import models
+
+
+def active_venues():
+    return models.Venue.objects.filter(active=True)
```
Comment (JobDoesburg) [inline website/venues/selectors.py:5]: Make this a Manager

**Review by JobDoesburg — CHANGES_REQUESTED** [review]
I doubt if this venue calendar thing is the best way to separate modules. I think it is best to have borrel depend on the venue calendar app, and add one of those nice filters to display the kind of reservations that can be made for this calendar and add the buttons for it.

**Review by JobDoesburg — CHANGES_REQUESTED** [review]

**Comment by JobDoesburg** [comment]
TODO: 

- [ ] Display form and non-form error messages correctly in the template
- [ ] Use correct success redirects and user-messages
- [ ] Improve page structure (menu items, buttons, etc)
- [ ] Implement the logic on when to use ReservationSubmission and ReservationUpdate views, when to show buttons and when to allow access to the view
- [ ] Implement something to check out submitted borrel lists as read only objects
- [ ] Deleting reservation items (hit the DELETE checkbox if 0 is entered)
- [ ] Fix the calendar (currently, nothing shows up in the calendar)
- [ ] Better link borrel and venue reservations
- [ ] Venue reservations page and explainer
- [ ] Give other people access to your borrel reservation
- [ ] Approval for changing reservations (if people change their reservation, this should be approved? Can they even change?)
- [ ] Notification emails

**Comment by KiOui** [comment]
When reserving a Venue, I get this error:
Schermafbeelding 2022-03-07 om 15 11 25

**Review by JobDoesburg — APPROVED** [review]

**Pull request merged**